modules/atmosphere/rendering/renderableatmosphere.cpp

```diff
@@ -546,4 +499,78 @@ void RenderableAtmosphere::updateAtmosphereParameters() {
     _deferredcaster->setHardShadows(_hardShadowsEnabled);
 }
 
+// Calculate atmosphere dimming coefficient
+void RenderableAtmosphere::setDimmingCoefficient(const glm::dmat4& modelTransform) {
+    // Calculate if the camera is in the atmosphere and if it is in the sunny region
+    glm::dvec3 cameraPos = global::navigationHandler->camera()->positionVec3();
+    glm::dvec3 planetPos = glm::dvec3(modelTransform * glm::dvec4(0.0, 0.0, 0.0, 1.0));
+    float cameraDistance = static_cast<float>(glm::distance(planetPos, cameraPos));
+    glm::dvec3 normalUnderCamera = glm::normalize(cameraPos - planetPos);
+    glm::dvec3 vecToSun = glm::normalize(-planetPos);
```

Comment:
This should have a comment that we assume the sun's position is (0, 0, 0). Although, it isn't (but don't tihnk it matters for these computations)

This would make it easier in the future, when we decide to fix all these errors we have in assuming that the Sun is in the origin.